packages/headless/src/api/analytics/product-listing-analytics.ts

```diff
@@ -0,0 +1,55 @@
+import {SearchPageClientProvider} from 'coveo.analytics';
+import {SearchEventRequest} from 'coveo.analytics/dist/definitions/events';
+import {
+  ConfigurationSection,
+  ProductListingSection,
+  SearchHubSection,
+} from '../../state/state-sections';
+import {getProductListingInitialState} from './../../features/product-listing/product-listing-state';
+import {BaseAnalyticsProvider} from './base-analytics';
+
+export type StateNeededByProductListingAnalyticsProvider =
+  ConfigurationSection & Partial<SearchHubSection & ProductListingSection>;
+
+export class ProductListingAnalyticsProvider
+  extends BaseAnalyticsProvider<StateNeededByProductListingAnalyticsProvider>
+  implements SearchPageClientProvider
+{
+  private initialState = getProductListingInitialState();
+
+  public getPipeline(): string {
+    return '';
+  }
+
+  public getSearchEventRequestPayload(): Omit<
+    SearchEventRequest,
+    'actionCause' | 'searchQueryUid'
+  > {
+    return {
+      queryText: '',
+      responseTime: 0,
+      results: this.mapResultsToAnalyticsDocument(),
+      numberOfResults: this.numberOfResults,
+    };
+  }
+
+  public getSearchUID() {
+    const newState = this.getState();
+    return newState.productListing?.responseId || this.initialState.responseId;
+  }
+
+  private mapResultsToAnalyticsDocument() {
+    return this.state.productListing?.products.map((p) => ({
+      documentUri: p.documentUri,
+      documentUriHash: p.documentUriHash,
```

Comment:
Is the `urihash` returned with the product listing results by default now?
I had to add it in `additionalFields` when I did it manually.

Reply:
I opened a PR for it, waiting to see how the team will react https://github.com/coveo/commerce-service/pull/915